### PR TITLE
Remove usages of `call_user_func_array()`

### PR DESCRIPTION
**Pull request by phansys**
|Q            |A     |
|---          |---   |
|Branch       |master|
|Bug fix?     |no    |
|New feature? |no    |
|BC breaks?   |no    |
|Deprecations?|no    |
|Tests pass?  |yes   |
|Fixed tickets|n/a   |
|License      |MIT   |
|Doc PR       |n/a   |

### CODE REVIEW

File: src/Admin/BaseFieldDescription.php [hunk 1]
```diff
@@ -297,14 +297,14 @@ public function getFieldValue($object, $fieldName)
...
-                return \call_user_func_array([$object, $getter], $parameters);
+                return $object->{$getter}(...$parameters);
             }
         }
 
         if (method_exists($object, '__call')) {
             $this->cacheFieldGetter($object, $fieldName, 'call');
 
-            return \call_user_func_array([$object, '__call'], [$fieldName, $parameters]);
+            return $object->__call($fieldName, $parameters);
```
Comment (greg0ire) [inline src/Admin/BaseFieldDescription.php]: Isn't that the same as `$object->$fieldname($parameters);` ?

Comment (phansys) [reply src/Admin/BaseFieldDescription.php]: I think that the answer is yes, but then why the previous call was not `\call_user_func_array([$object, $fieldName], $parameters);`?
Note that the check is not `method_exists($object, $fieldName)`.

File: src/Controller/CRUDController.php [hunk 2]
```diff
@@ -79,7 +79,7 @@ class CRUDController implements ContainerAwareInterface
     public function __call($method, $arguments)
     {
         if (\in_array($method, ['get', 'has'], true)) {
-            return \call_user_func_array([$this->container, $method], $arguments);
+            return $this->container->{$method}(...$arguments);
```
Comment (greg0ire) [inline src/Controller/CRUDController.php]:
```suggestion
            return $this->container->$method(...$arguments);
```

File: src/Controller/PolyfillControllerTrait.php [hunk 3]
```diff
@@ -61,7 +61,7 @@ public function __construct(ContainerInterface $container)
 
     public function proxyCall($method, $arguments)
     {
-        return \call_user_func_array([$this, $method], $arguments);
+        return $this->{$method}(...$arguments);
```
Comment (greg0ire) [inline src/Controller/PolyfillControllerTrait.php]:
```suggestion
        return $this->$method(...$arguments);
```

File: src/Admin/BaseFieldDescription.php (hunk 1 above)
Comment (phansys) [reply src/Admin/BaseFieldDescription.php]: I think `method_exists($object, $fieldName)` will return `true` even if the method is not publicly accessible.

Comment (greg0ire) [reply src/Admin/BaseFieldDescription.php]: yeah, but `$fieldName` is not a method name, is it? It's the name of a field, right?

Comment (phansys) [reply src/Admin/BaseFieldDescription.php]: I await for your comments: https://3v4l.org/16GNR.

Comment (greg0ire) [reply src/Admin/BaseFieldDescription.php]: Isn't this a more realistic scenario of would happen here: https://3v4l.org/Bqn9Z ?

Comment (phansys) [reply src/Admin/BaseFieldDescription.php]: I guess, so I think we should preserve the call to `__call()` in order to delegate the responsibility to its implementation.

**Comment by OskarStark** [comment]
Can you please write why we should move away from this usage in the PR header?

File: src/Admin/BaseFieldDescription.php (hunk 1 above)
Comment (greg0ire) [reply src/Admin/BaseFieldDescription.php]: I really don't see the point of keeping `__call` when you could write `$fieldName`, but as you wish…

**Comment by phansys** [comment]
> Can you please write why we should move away from this usage in the PR header?

The main reason for these changes is to avoid 2 unrequired situations:
* Call to `call_user_func()` / `call_user_func_array()`;
* Create a dummy array just to allow these functions to work.

**Pull request merged**